scripts/biomedical/VirtualMetabolicHuman/format_metabolite.py

```diff
@@ -0,0 +1,475 @@
+    except:
+        meta_cyc = 'NA'
+    try:
+        bigg = elem.xpath('hmdb:bigg_id/text()', namespaces=ns)[0]
+    except:
+        bigg = 'NA'
+    try:
+        metlin_id = elem.xpath('hmdb:metlin_id/text()', namespaces=ns)[0]
+    except:
+        metlin_id = 'NA'
+    try:
+        pdb_id = elem.xpath('hmdb:pdb_id/text()', namespaces=ns)[0]
+    except:
+        pdb_id = 'NA'
+    try:
+        logpexp = elem.xpath('hmdb:experimental_properties/hmdb:property[hmdb:kind = "logp"]/hmdb:value/text()', namespaces=ns)[0]
+    except:
+        logpexp = 'NA'
+    try:
+        kingdom = elem.xpath('hmdb:taxonomy/hmdb:kingdom/text()', namespaces=ns)[0]
+    except:
+        kingdom = 'NA'
+    try:
+        direct_parent = elem.xpath('hmdb:taxonomy/hmdb:direct_parent/text()', namespaces=ns)[0]
+    except:
+        direct_parent = 'NA'
+    try:
+        super_class = elem.xpath('hmdb:taxonomy/hmdb:super_class/text()', namespaces=ns)[0]
+    except:
+        super_class = 'NA'
+    try:
+        classorg = elem.xpath('hmdb:taxonomy/hmdb:class/text()', namespaces=ns)[0]
+    except:
+        classorg = 'NA'
+    try:
+        sub_class = elem.xpath('hmdb:taxonomy/hmdb:sub_class/text()', namespaces=ns)[0]
+    except:
+        sub_class = 'NA'
+    try:
+        molecular_framework = elem.xpath('hmdb:taxonomy/hmdb:molecular_framework/text()', namespaces=ns)[0]
+    except:
+        molecular_framework = 'NA'
+    try:
+        vmh_id = elem.xpath('hmdb:vmh_id/text()', namespaces=ns)[0]
+    except:
+        vmh_id = 'NA'    
+
+    writer.writerow({'accession': accession, 'monisotopic_molecular_weight': monisotopic_molecular_weight, 'iupac_name': iupac_name, 'name': name, 'chemical_formula': chemical_formula, 'InChIKey': inchikey, 'cas_registry_number': cas_registry_number, 'smiles': smiles,'drugbank': drugbank,'chebi_id': chebi_id,'pubchem': pubchem,'phenol_explorer_compound_id':phenol_explorer_compound_id, 'food': food,'knapsack': knapsack, 'chemspider': chemspider,'kegg': kegg, 'meta_cyc': meta_cyc, 'bigg':bigg, 'metlin_id': metlin_id, 'pdb_id':pdb_id,'logpexp':logpexp, 'kingdom': kingdom, 'direct_parent': direct_parent, 'super_class': super_class, 'class': classorg, 'sub_class': sub_class, 'molecular_framework': molecular_framework, 'vmh_id': vmh_id})
+    # It's safe to call clear() here because no descendants will be
+    # accessed
+    elem.clear()
+    # Also eliminate now-empty references from the root node to elem
+    for ancestor in elem.xpath('ancestor-or-self::*'):
+        while ancestor.getprevious() is not None:
+            del ancestor.getparent()[0]
+  del context
+  return;
+
+#Used to get hmdb id for common name matches between vmh and hmdb
+def name_map(dfm, dfh):
+    dfm['fullName'] = dfm['fullName'].map(lambda x: re.sub(r'\W+', '', x))
+    dfm['fullName'] = dfm['fullName'].str.lower()
+    dfh['name'] = dfh['name'].map(lambda x: re.sub(r'\W+', '', x))
```

Comment:
Nit: It would also work (and possibly more efficient) to apply all the transformations at once in a single lambda / helper function.

Reply:
Done !